packages/grid/src/vaadin-grid-sorter.js

```diff
@@ -210,11 +210,14 @@ class GridSorter extends ThemableMixin(DirMixin(PolymerElement)) {
 
     this.dispatchEvent(
       new CustomEvent('sorter-changed', {
-        detail: { shiftClick: this._shiftClick },
+        detail: { shiftClick: this._shiftClick, fromSorterClick: this._fromSorterClick },
         bubbles: true,
         composed: true,
       }),
     );
+    // Cleaning up as a programatically sorting can be done after some user interaction
+    this._fromSorterClick = false;
```

Comment:
Please add a test case for this scenario

Reply:
Added a couple of tests for both flags.